File: week_2/day_08_stack/variations_que/evaluate_postfix.py
# evaluate_postfix.py
def evalPostfix(tokens):
    stack = []
    for t in tokens:
        if t in "+-*/":
            b, a = stack.pop(), stack.pop()
            stack.append(int(eval(f"{a}{t}{b}")))
        else:
            stack.append(int(t))
    return stack[0]

# INPUT
print(evalPostfix(["2","1","+","3","*"]))  # 9

File: week_2/day_08_stack/variations_que/evaluatePostfix.js
// evaluatePostfix.js
function evalPostfix(tokens) {
    const stack = [];
    for (let t of tokens) {
        if ("+-*/".includes(t)) {
            const b = stack.pop(), a = stack.pop();
            stack.push(Math.trunc(eval(a + t + b)));
        } else {
            stack.push(Number(t));
        }
    }
    return stack[0];
}

// INPUT
console.log(evalPostfix(["2","1","+","3","*"])); // 9
